src/google/protobuf/stubs/common.h

```diff
@@ -88,20 +88,20 @@ namespace internal {
 
 // The minimum library version which works with the current version of the
 // headers.
-#define GOOGLE_PROTOBUF_MIN_LIBRARY_VERSION 3006001
+#define GOOGLE_PROTOBUF_MIN_LIBRARY_VERSION GOOGLE_PROTOBUF_VERSION
```

Comment:
https://github.com/protocolbuffers/protobuf/commit/050fc9a43717193ffbccbd8ca5d19e1e5648c815#diff-cbd31ea2a5761cd281396b4277c36887
I think they are not necessarily always the same.

Reply:
For minor releases, we usually just update ```GOOGLE_PROTOBUF_VERSION```

Reply:
The documentation seems to suggest that we should update all, but reverted this file, and let the script updates all of them by default. The author can change the version number manually if necessary.

Reply:
I think the document was written for major releases.
Is this script also used for major release only?

Reply:
If it's minor release, I think we should keep things like ```GOOGLE_PROTOBUF_MIN_LIBRARY_VERSION``` unchanged. Because the minor release has no API change and the new library should work with the old header and the new header should also work with the old library.

Reply:
The script simply calculate the version based on the given version number:
https://github.com/protocolbuffers/protobuf/pull/5454/commits/79748f318442c70cab077ac44da92c8bd84eb077. So if the given version is 3.6.1, it will generate 3006001.

Should I revise it such that only GOOGLE_PROTOBUF_VERSION would care about the micro version, and all other constants would still treat it like 3006000?

Reply:
Done. Revised script to only update these constants on major release.